### ISSUE TITLE
gamelist.hpp:398:39: error: ‘PossiblyYes’ is not a member of ‘camoto::gamegraphics::TilesetType’

### ISSUE THREAD

**Issue by owenmpierce** [Malvineous/camoto-studio#4]
I tried to build from sources, but I couldn't build camoto-studio.

First of all, libgame\* requires boost >= 1.59, [but your website says boost >=1.46](http://www.shikadi.net/camoto#Build_instructions).  I also needed to install package "xmlto," and  ./configure was unable to find my boost libraries, even though I'd installed them to /usr/local/lib.  I was able to get around this by supplying --with-boost-libdir=/usr/local/lib to the configure script.

Anyways, when I tried to build camoto-studio, after successfully building and installing the prerequisite libgame libraries, here's what I got:

```
./configure 
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for g++... g++
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking for style of include used by make... GNU
checking dependency style of g++... gcc3
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking how to print strings... printf
checking for gcc... gcc
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking for a sed that does not truncate output... /bin/sed
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for fgrep... /bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for mt... mt
checking if mt is a manifest tool... no
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /usr/bin/ld -m elf_x86_64
checking if the linker (/usr/bin/ld -m elf_x86_64) is GNU ld... yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... yes
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether g++ supports C++11 features by default... no
checking whether g++ supports C++11 features with -std=c++11... yes
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for libgamecommon... yes
checking for libgamearchive... yes
checking for libgamegraphics... yes
checking for libgamemaps... yes
checking for libgamemusic... yes
checking for libxml2... yes
checking for gtk... yes
checking for portaudio... yes
checking for libpng... yes
checking for png++... yes
checking if png++ is new enough... yes
checking whether make supports nested variables... (cached) yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating doc/Makefile
config.status: creating config.h
config.status: config.h is unchanged
config.status: executing depfiles commands
config.status: executing libtool commands

```

```
 make
make  all-recursive
make[1]: Entering directory `/home/owen/sources/camoto-studio'
Making all in src
make[2]: Entering directory `/home/owen/sources/camoto-studio/src'
  CXX      main.o
In file included from main.hpp:35:0,
                 from main.cpp:37:
gamelist.hpp: In instantiation of ‘decltype ((Type*)(nullptr)->.open(std::move(content), suppData)) openObject(Gtk::Window*, const GameObject&, std::unique_ptr<camoto::stream::inout>, camoto::SuppData&, DepData*, Project*) [with Type = camoto::gamegraphics::TilesetType; decltype ((Type*)(nullptr)->.open(std::move(content), suppData)) = std::shared_ptr<camoto::gamegraphics::Tileset>; camoto::SuppData = std::map<camoto::SuppItem, std::unique_ptr<camoto::stream::inout> >; DepData = std::map<DepType, std::unique_ptr<GameObjectInstance> >]’:
main.cpp:186:28:   required from here
gamelist.hpp:398:39: error: ‘PossiblyYes’ is not a member of ‘camoto::gamegraphics::TilesetType’
  if (fmtHandler->isInstance(*content) < Type::PossiblyYes) {
                                       ^
make[2]: *** [main.o] Error 1
make[2]: Leaving directory `/home/owen/sources/camoto-studio/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/owen/sources/camoto-studio'
make: *** [all] Error 2

```

Is there something I did wrong?  

Thanks.

**Comment by Malvineous** [comment]
Sorry it's in a bit of flux at the moment.  I updated the whole project for C++11 but only the libgame\* libraries are done, so if you want to build `camoto-studio` you'll need version 1.x of all the libgame\* libraries instead, as well as building `camoto-studio` from the `1.x` branch.  The `master` branch of `camoto-studio` matches v2.x of the libgame\* libraries (their `master` branches are also at v2.x).

So basically build everything from the `1.x` branch if you want the old GUI which sort of works, or for the latest code build the `master` branches of everything except `camoto-studio` because `camoto-studio/master` can't really do anything yet.

**Comment by owenmpierce** [comment]
Okay, thanks for the reply.
